modules/packages/MPI.chpl

```diff
@@ -38,37 +122,123 @@ since all of these are built around user-defined handlers, that we support.
      in which case we write it using an array form.
   #. Some MPI-1.1 functions were deprecated in MPI-2. These should be updated in the future, but
      are still present in this version.
-  #. We represent MPI_Aint by ptrdiff. If this is not the correct size, there will be an assertion 
+  #. We represent MPI_Aint by ptrdiff. If this is not the correct size, there will be an assertion
      failure in the code.
 */
 module MPI {
   use SysCTypes;
   require "mpi.h";
 
-  /* Automatically initialize, and define
-   worldSize and worldRank, since we will
-   likely use those often. 
+  use UtilReplicatedVar;
 
-   You are still responsible for calling MPI_Finalize,
-   until Chapel has a module termination scheme.
-   */
+  /* Configuration constants */
   config const autoInitMPI=true;
+  config const requireThreadedMPI=true;
+  config const debugMPI=false;
+
+  /* Define a new communicator that directly maps to Chapel locales.
+  This is just a reordering of MPI_COMM_WORLD, which you are, of course,
+  free to continue to use. This just guarantees that locale ids and MPI
+  ranks agree.
+
+  Since this is a replicated variable, you must use CHPL_COMM_WORLD(1)
+  in your code.
```

Comment:
You could allow users to use CHPL_COMM_WORLD directly if you:
 * made the replicated variable be e.g. CHPL_COMM_WORLD_REPLICATED
 * created a parethesisless function returning CHPL_COMM_WORLD_REPLICATED(1) like this:

      proc CHPL_COMM_WORLD {
           return CHPL_COMM_WORLD_REPLICATED(1);
     }

There are other ways to create replicated variables - especially global ones - and so I'd prefer not to have this detail exposed in the user-facing API. The parenthesis-less function should work around it while allowing you to implement it in the same way.